### PR TITLE
Feature: approve screen

### LINKED ISSUE #1218: Update approve screen
Per new designs for the extension, we should update the Approve screen on mobile as well. 

Designs are [here](https://www.figma.com/file/Emfl6CRPRtuQtzJEOJJ3hy/Approve-Screen?node-id=606%3A0)

Changes to other transaction confirmation screens can be handled in a separate issue.

### LINKED ISSUE #1243: Android: Approves waaaay more DAI than it should
Tried buying a sponsorship from [sp.brightid.org](https://sp.brightid.org) from the metamask mobile app today, and noticed that it wanted me to approve 10^18 DAI :grimacing: 

Needless to say, I rejected the transaction.

At first I thought it was an issue with their code, but when I tried it on my desktop it behaved as expected.

Here is the relevant code snippet.

```
dai = parseFloat($("#spDai").val()) * 10 ** 18;
  ptContract.approve.sendTransaction(addresses.sp_minter, dai, function (error, result) {}
```

On desktop it asks to approve 1DAI
Screenshot from 2019-12-10 10-19-40

On mobile it asks to approve 1,000,000,000,000,000,000 DAI
image

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

Updated ERC20 approve screen per https://www.figma.com/file/Emfl6CRPRtuQtzJEOJJ3hy/Approve-Screen?node-id=721%3A1

I reused some components from https://github.com/MetaMask/metamask-mobile/pull/1239 so it would be a good idea to merge that one first.

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #1218
Resolves #1243

### CODE REVIEW

File: app/components/Views/SendFlow/CustomGas/index.js [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+	convertApiValueToGWEI,
+	parseWaitTime
+} from '../../../../util/custom-gas';
+import { BN } from 'ethereumjs-util';
+import { fromWei, renderWei, hexToBN, renderFromWei, isBN, isDecimal } from '../../../../util/number';
+import Logger from '../../../../util/Logger';
+import { getTicker } from '../../../../util/transactions';
+import Ionicon from 'react-native-vector-icons/Ionicons';
+
+const AVERAGE_GAS = 20;
```
Comment (brunobar79) [inline app/components/Views/SendFlow/CustomGas/index.js]: I believe I've added the same comment in the Address book / sendflow PR. Time to move it to constants

File: app/styles/common.js [hunk 2]
```diff
@@ -15,18 +15,21 @@ export const colors = {
 	black: '#000000',
 	white: '#FFFFFF',
 	white100: '#F9FAFB',
+	grey600: '#5B5D67',
 	grey500: '#6a737d',
 	grey400: '#848c96',
 	grey300: '#9fa6ae',
 	grey200: '#bbc0c5',
 	grey100: '#d6d9dc',
+	grey050: '#D8D8D8',
```
Comment (brunobar79) [inline app/styles/common.js]: lmao

File: scripts/build.sh [hunk 3]
```diff
@@ -163,7 +163,7 @@ buildIosRelease(){
 		if [ ! -f "ios/release.xcconfig" ] ; then
 			echo $IOS_ENV | tr "|" "\n" > ios/release.xcconfig
 		fi
-		./node_modules/.bin/react-native run-ios  --configuration Release --simulator "iPhone 11 Pro (13.2)"
+		./node_modules/.bin/react-native run-ios  --configuration Release --simulator "iPhone 11 Pro (13.2.2)"
```
Comment (brunobar79) [inline scripts/build.sh]: 13.3?

**Review by brunobar79 — APPROVED** [review]
This is great. Good stuff @estebanmino

**Comment by ibrahimtaveras00** [comment]
A majority of the issues are fixed

Outstanding issue I am seeing is when I edit the gas amount to slow, and then edited the permission from unlimited to custom spend max. 

On Android I get this warning:

image

On iOS I get this error:

image

**Comment by ibrahimtaveras00** [comment]
Only outstanding issue = custom spend limit value should be a whole number and not a decimal

**Review by ibrahimtaveras00 — APPROVED** [review]
All issues have been resolved on both OS's, QA passed 👍

**Pull request merged**